### ISSUE TITLE
Pipeline for face recognition

### ISSUE THREAD

**Issue by NeonParticlee** [NeonParticlee/Voice-recognition-#17]
Build training pipeline for the facial recognition system

**Comment by Rogue-47** [comment]
The training pipeline has not been fully coded but the functionalities needed for implementing it are almost done:
here's exactly what is finished:
- The forward function (to receive the image(s) and train or forward to classify the face for inference sessions)
- The loss function (triplet loss was chosen)
- The back propagation functionality (to calculate the gradients of the images data)
- ADAM (Adaptive moment estimator) optimizer (the kernel launcher (the function that runs it on the GPU) for this is only finished but not everything is done
what's not finished yet?
- need to implement a function that handles the initialization of all the parameters (in a parallel way) that will be fed to the network (weights, biases, gradient weights, gradient biases, momentum of the gradients, second moment of the gradients, etc.)
- need to implement a randomizer to randomly select anchor-negative images from the dataset for triplet loss
- formatted csv file writing at the end of the session is necessary to store the embedded features of the images and the names of people for the model's inference sessions later.
- adding other functionality for the ADAM optimizer and experimenting with different GPU kernel launch parameters (like grid size and block size).
